libbeat/esclientleg/connection.go

```diff
@@ -0,0 +1,258 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package esclientleg
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/elastic/beats/libbeat/outputs/transport"
```

Comment:
I don't like how the `esclientleg` package here has a dependency on a package under `libbeat/outputs`. I wonder if we should move `libbeat/outputs/transport` to `libbeat/transport`. WDYT @urso?

Reply:
If you agree, I think we should make this change in a separate PR (which would be a pre-requisite to this PR here) since this PR here is getting quite large already.

Reply:
the libbeat/outputs/transport package is legacy and has been superseded by libbeat/common/transport. Right now 'transport' only contains TLS support, but I think it would be nice to have some more common client/networking setup functionality. E.g. to unify common settings, setup, initialization in outputs, inputs, and modules.